test/tests/Pathfinding.cpp

```diff
@@ -0,0 +1,237 @@
+            uint8_t pathingResult = 0;
+            peep->PerformNextAction(pathingResult);
+            ++step;
+
+            pos->x = peep->x / 32;
+            pos->y = peep->y / 32;
+            pos->z = peep->z / 8;
+
+            EXPECT_PRED_FORMAT1(AssertIsNotForbiddenPosition, *pos);
+        }
+        
+        // Clean up the peep, because we're reusing this loaded context for all tests.
+        peep_sprite_remove(peep);
+
+        // Require that the number of steps taken is exactly what we expected. The pathfinder is supposed to be
+        // deterministic, and we reset the RNG seed for each test, everything should be entirely repeatable; as
+        // such a change in the number of steps taken on one of these paths needs to be reviewed.
+        EXPECT_EQ(step, maxSteps);
+
+        return *pos == goal;
+    }
+
+    static ::testing::AssertionResult AssertIsStartPosition(const char*, const TileCoordsXYZ& location)
+    {
+        return AssertPositionIsSetUp("Start", 11u, location);
+    }
+
+    static ::testing::AssertionResult AssertIsGoalPosition(const char*, const TileCoordsXYZ& location)
+    {
+        return AssertPositionIsSetUp("Goal", 9u, location);
+    }
+
+    static ::testing::AssertionResult AssertIsNotForbiddenPosition(const char*, const TileCoordsXYZ& location)
+    {
+        const uint32_t forbiddenSurfaceStyle = 8u;
+
+        const uint32_t style = map_get_surface_element_at(location.x, location.y)->AsSurface()->GetSurfaceStyle();
+
+        if (style == forbiddenSurfaceStyle)
+            return ::testing::AssertionFailure()
+                << "Path traversed location " << location << ", but it is marked as a forbidden location (surface style "
+                << forbiddenSurfaceStyle << "). Either the map is set up incorrectly, or the pathfinder went the wrong way.";
+
+        return ::testing::AssertionSuccess();
+    }
+
+private:
+
+    static ::testing::AssertionResult AssertPositionIsSetUp(const char* positionKind, uint32_t expectedSurfaceStyle, const TileCoordsXYZ& location)
+    {
+        const uint32_t style = map_get_surface_element_at(location.x, location.y)->AsSurface()->GetSurfaceStyle();
+
+        if (style != expectedSurfaceStyle)
+            return ::testing::AssertionFailure()
+                << positionKind << " location " << location << " should have surface style " << expectedSurfaceStyle
+                << " but actually has style " << style
+                << ". Either the test map is not set up correctly, or you got the coordinates wrong.";
+
+        return ::testing::AssertionSuccess();
+    }
+
+    static std::shared_ptr<IContext> _context;
+};
+
+std::shared_ptr<IContext> PathfindingTestBase::_context;
+
+struct SimplePathfindingScenario
+{
+    const char* name;
+    TileCoordsXYZ start;
+    TileCoordsXYZ goal;
+    uint32_t steps;
+
+    SimplePathfindingScenario(const char* _name, const TileCoordsXYZ& _start, const TileCoordsXYZ& _goal, int _steps = 10000)
```

Comment:
Yeah this is a holdover from before I realised it would make sense to expect a specific number of steps. I don't specify the value for the negative tests right now, but it's easy enough to change that and eliminate the default here.